### PR TITLE
Add filtering to collections

### LINKED ISSUE #1171: Add filtering capability to collection browser
I'm going through the Bacillus collection and it would be nice to be able to filter by a particular part type so you could easily pull out all the promoters, CDS or terminators.

### PR DESCRIPTION
**Pull request by cjmyers**
Closes #1171 

Adds filtering to collections

### CODE REVIEW

File: templates/views/collection.jade [hunk 1]
```diff
@@ -30,13 +30,24 @@ block modal
 
 block topLevelPreview
     h3 Members
+    if types
+        form(name="filterTypes", method="post").submission-form
```
Comment (3ach) [inline templates/views/collection.jade:34]: This form looks squished together. Should be a proper button addon as shown [here](https://getbootstrap.com/docs/3.3/components/#input-groups-buttons).

File: browser/synbiohub.js [hunk 2]
```diff
@@ -128,8 +128,11 @@ if (typeof meta !== 'undefined') {
         serverSide: true,
 
       searching: !meta.remote,
+      order: [[0, "asc"]],
+      "columnDefs": [
```
[the remark this answers is not in the document]
Comment (cjmyers) [reply browser/synbiohub.js]: Changed

File: templates/views/collection.jade (hunk 1 above)
Comment (cjmyers) [reply templates/views/collection.jade:34]: @3ach I'm not sure if I've resolved this one that way you were thinking.  Could you check and see if there was something else you had in mind?

**Comment by cjmyers** [comment]
@3ach I've gone ahead and merged this, but if you have ideas how to make it look better, let me know.

**Pull request merged**